client/src/views/routes/guest-route.js

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Navigate, Outlet } from "react-router-dom";
+import useAuth from "../../hooks/useAuth";
+
+const GuestRoute = () => {
+    const { user, token } = useAuth();
+
+    if (user !== "null" && token !== "null") {
```

Comment:
Ở đây không cần viết như vậy user !== "null" có thể viết tắt lại thành !user. Thứ 2 là user vs token nó đi song song với nhau. Token mang toàn bộ thông tin của user chỉ có điều được mã hóa thôi nên chỉ cần 1 trong 2 chứ không cần cả 2 đâu nhae